### PR TITLE
Update service urls for existing events

### PR DESCRIPTION
**Pull request by akravch**
Add ServiceUrls table.
Update service url to be used to send a message when a new user message comes in.
Keep the history of service url usages in ScheduledMessageDetailsServiceUrls table.

#20

### CODE REVIEW

File: SchedulerBot/SchedulerBot.Database.Entities/ScheduledMessageDetails.cs [hunk 1]
```diff
@@ -61,5 +57,11 @@ public class ScheduledMessageDetails
 		/// Gets or sets the UTC offset of the recipient time zone.
 		/// </summary>
 		public TimeSpan? TimeZoneOffset { get; set; }
+
+		/// <summary>
+		/// Gets or sets the collection of <see cref="ScheduledMessageDetailsServiceUrl"/> objects
+		/// connecting the current insance to corresponding <see cref="ServiceUrl"/> objects.
```
Comment (pryabov) [inline SchedulerBot/SchedulerBot.Database.Entities/ScheduledMessageDetails.cs]: `insance` -> `instance`

File: SchedulerBot/SchedulerBot.Database.Entities/ScheduledMessageDetailsServiceUrl.cs [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+{
+	/// <summary>
+	/// Enables many-to-many relationship between <see cref="ScheduledMessageDetails"/> and <see cref="ServiceUrl"/>.
+	/// </summary>
+	public class ScheduledMessageDetailsServiceUrl
+	{
+		/// <summary>
+		/// Gets or sets the scheduled message details identifier.
+		/// </summary>
+		public Guid DetailsId { get; set; }
```
Comment (pryabov) [inline SchedulerBot/SchedulerBot.Database.Entities/ScheduledMessageDetailsServiceUrl.cs]: `DetailsId` possibly rename

File: SchedulerBot/SchedulerBot.Database.Entities/ScheduledMessageDetailsServiceUrl.cs [hunk 3]
```diff
@@ -0,0 +1,35 @@
+﻿using System;
+
+namespace SchedulerBot.Database.Entities
+{
+	/// <summary>
+	/// Enables many-to-many relationship between <see cref="ScheduledMessageDetails"/> and <see cref="ServiceUrl"/>.
+	/// </summary>
+	public class ScheduledMessageDetailsServiceUrl
```
Comment (pryabov) [inline SchedulerBot/SchedulerBot.Database.Entities/ScheduledMessageDetailsServiceUrl.cs]: Add `ICreatedOn` interface?

File: SchedulerBot/SchedulerBot.Database.Entities/ScheduledMessageDetails.cs (hunk 1 above)
Comment (akravch) [reply SchedulerBot/SchedulerBot.Database.Entities/ScheduledMessageDetails.cs]: Good catch, I finally installed ReSpeller after that :D

File: SchedulerBot/SchedulerBot.Database.Entities/ScheduledMessageDetailsServiceUrl.cs (hunk 2 above)
Comment (akravch) [reply SchedulerBot/SchedulerBot.Database.Entities/ScheduledMessageDetailsServiceUrl.cs]: Hm, not sure I am following.

It specifies the ID of ScheduledMessageDetails, which is pretty much `DetailsId`. Another thing is that the `DetailId` is actually also the ID of ScheduledMessage, i.e. it is the same that `MessageId`.

Is that what you've meant?

File: SchedulerBot/SchedulerBot.Database.Entities/ScheduledMessageDetailsServiceUrl.cs (hunk 3 above)
Comment (akravch) [reply SchedulerBot/SchedulerBot.Database.Entities/ScheduledMessageDetailsServiceUrl.cs]: Did it, although there is no real usage for that for now.

File: SchedulerBot/SchedulerBot.Database.Entities/ScheduledMessageDetailsServiceUrl.cs (hunk 2 above)
Comment (pryabov) [reply SchedulerBot/SchedulerBot.Database.Entities/ScheduledMessageDetailsServiceUrl.cs]: Idea was to give a name `ScheduledMessageDetailId`

Comment (akravch) [reply SchedulerBot/SchedulerBot.Database.Entities/ScheduledMessageDetailsServiceUrl.cs]: @pryabov We could do that, but I think then we would have to rename `Details` property to `ScheduledMessageDetails` for consistency.

Do you think we should do this?

**Pull request merged**